task/backend/scheduler.go

```diff
@@ -337,20 +325,33 @@ func (s *TickScheduler) UpdateTask(task *StoreTask, meta *StoreTaskMeta) error {
 	}
 	ts.task = task
 
-	next, err := meta.NextDueRun()
+	next, err := s.taskControlService.NextDueRun(authCtx, task.ID)
 	if err != nil {
 		return err
 	}
-	hasQueue := len(meta.ManualRuns) > 0
+
+	runs, err := s.taskControlService.ManualRuns(authCtx, task.ID)
+	if err != nil {
+		return err
+	}
+
+	hasQueue := len(runs) > 0
 	// update the queued information
 	ts.nextDueMu.Lock()
 	ts.hasQueue = hasQueue
 	ts.nextDue = next
+	ts.authCtx = authCtx
 	ts.nextDueMu.Unlock()
-
 	// check the concurrency
 	// todo(lh): In the near future we may not be using the scheduler to manage concurrency.
-	maxC := int(meta.MaxConcurrency)
+	opt, err := options.FromScript(task.Flux)
```

Comment:
Since `options.FromScript` could fail, could that be moved earlier in the method so we can avoid some locking if it happens to produce an error?